pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/participant/ParticipantWrapper.java

```diff
@@ -114,52 +106,65 @@ private List<ParticipantWrapperDto> collectData(DDPInstanceDto ddpInstanceDto) {
         List<String> proxyGuids = new ArrayList<>();
 
         for (ElasticSearchParticipantDto elasticSearchParticipantDto : esData.getEsParticipants()) {
+            if (elasticSearchParticipantDto instanceof UnparsedESParticipantDto) {
+                addWrapperToList((UnparsedESParticipantDto) elasticSearchParticipantDto, result);
+            } else {
+                addWrapperToList(elasticSearchParticipantDto, result);
+            }
+            proxyGuids.addAll(elasticSearchParticipantDto.getProxies());
+        }
+        fillParticipantWrapperDtosWithProxies(result, proxyGuids);
+        return result;
+    }
 
-            elasticSearchParticipantDto.getDsm().ifPresent(esDsm -> {
-
-                Participant participant = esDsm.getParticipant().orElse(new Participant());
+    private void addWrapperToList(UnparsedESParticipantDto elasticSearchParticipantDto, List<ParticipantWrapperDto> result) {
+        // don't do any copying of the main attributes, but do keep track of the proxies so that we can fetch them in bulk
+        ParticipantWrapperDto participantWrapperDto = new ParticipantWrapperDto();
+        participantWrapperDto.setEsData(elasticSearchParticipantDto);
+        result.add(participantWrapperDto);
+    }
 
-                esDsm.getOncHistory().ifPresent(oncHistory -> {
-                    participant.setCreated(oncHistory.getCreated());
-                    participant.setReviewed(oncHistory.getReviewed());
-                });
+    private void addWrapperToList(ElasticSearchParticipantDto elasticSearchParticipantDto, List<ParticipantWrapperDto> result) {
 
-                StudyPreFilter.fromPayload(StudyPreFilterPayload.of(elasticSearchParticipantDto, ddpInstanceDto))
-                        .ifPresent(StudyPreFilter::filter);
```

Comment:
I think this is necessary for studies which have different versions, @Anzori-Ghurtchumelia can help about this

Reply:
ah, great catch, I didn't mean to remove this -- it must have gotten accidentally deleted in a merge